### PR TITLE
Dialog component css version

### PR DESCRIPTION
**Pull request by stevedya**
Draft: Example of #8513 using with raw css with tailwinds theme function. This PR for comparison purposes.

### CODE REVIEW

**Comment by lb-** [comment]
I like this a lot

File: wagtail/admin/templates/wagtailadmin/shared/dialog/dialog.html [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{% load wagtailadmin_tags i18n %}
+{% with id=title|cautious_slugify %}
+    <template data-dialog>
+        <div
+            id="{{ id }}"
+            aria-labelledby="title-{{ id }}"
+            aria-hidden="true"
+            class="{{ class_name }} dialog"
```
Comment (thibaudcolas) [inline wagtail/admin/templates/wagtailadmin/shared/dialog/dialog.html:8]: I’d prefer we don’t support arbitrary classes. If we want to create variations of the dialog component then they should have a proper API rather than relying on class names.

```suggestion
            class="dialog"
```

File: wagtail/admin/templatetags/wagtailadmin_tags.py [hunk 2]
```diff
@@ -893,3 +892,72 @@ def component(context, obj, fallback_render_method=False):
...
+
+@register.inclusion_tag("wagtailadmin/shared/dialog/dialog.html")
+def dialog(
+    title,
+    icon_name=None,
+    subtitle=None,
+    message_status=None,
+    message_heading=None,
+    message_description=None,
+    class_name=None,
```
Comment (thibaudcolas) [inline wagtail/admin/templatetags/wagtailadmin_tags.py:905]:
```suggestion
```

File: client/scss/components/_dialog.scss [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    padding: theme('spacing.5');
+
+    &--info {
+      background: theme('colors.info.50');
+      color: theme('colors.info.100');
+    }
+
+    &--warning {
+      background: theme('colors.warning.50');
+      color: theme('colors.warning.100');
```
Comment (thibaudcolas) [inline client/scss/components/_dialog.scss:121]:
```suggestion
      color: theme('colors.primary.DEFAULT');
```

File: client/scss/components/_dialog.scss [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+  &__message-icon {
+    width: theme('spacing.5');
+    height: theme('spacing.5');
+    flex-shrink: 0;
+  }
+
+  &__message-header {
+    margin-inline-start: theme('spacing[2.5]');
+    padding-inline-end: theme('spacing.8');
+    color: theme('colors.grey.600');
```
Comment (thibaudcolas) [inline client/scss/components/_dialog.scss:144]:
```suggestion
    color: theme('colors.primary.DEFAULT');
    font-size: theme('fontSize.14');
```

File: client/scss/components/_dialog.scss [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+  }
+
+  &__message-heading {
+    font-size: theme('fontSize.14');
+    color: theme('colors.grey.600');
+  }
```
Comment (thibaudcolas) [inline client/scss/components/_dialog.scss:150]:
```suggestion
```

File: client/scss/components/_dialog.scss [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+  &__message-description {
+    margin-bottom: 0;
+    font-size: theme('fontSize.14');
+    color: theme('colors.grey.600');
```
Comment (thibaudcolas) [inline client/scss/components/_dialog.scss:155]:
```suggestion
```

File: wagtail/admin/templates/wagtailadmin/shared/dialog/dialog.html [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+                    {% icon name='cross' class_name="dialog__close-icon" %}
+                </button>
+
+                {% if message_heading and message_icon_name %}
+                    <div class="dialog__message dialog__message--{{ message_status }}">
+                        {% if message_icon_color %}
+                            {% icon name=message_icon_name class_name="dialog__message-icon "|add:message_icon_color %}
+                        {% else %}
+                            {% icon name=message_icon_name class_name="dialog__message-icon" %}
+                        {% endif %}
```
Comment (thibaudcolas) [inline wagtail/admin/templates/wagtailadmin/shared/dialog/dialog.html:22]:
```suggestion
                        {% icon name=message_icon_name class_name="dialog__message-icon" %}
```

**Review by thibaudcolas — CHANGES_REQUESTED** [review]

File: client/src/components/Sidebar/Sidebar.scss [hunk 8]
```diff
@@ -99,9 +99,9 @@
 
 // This is a separate component as it needs to display in the header
 .sidebar-nav-toggle {
+  @apply w-z-sidebar-toggle;
```
Comment (lb-) [inline client/src/components/Sidebar/Sidebar.scss:102]: Should we put these sidebar changes into a separate PR?

File: client/tailwind.config.js [hunk 9]
```diff
@@ -81,7 +81,19 @@ module.exports = {
           'left, inset-inline-start, padding-inline-start, width, transform, margin-top, min-height',
       },
       zIndex: {
-        header: '100',
+        'header': '100',
+        'sidebar': '110',
+        'sidebar-toggle': '120',
+        'dialog': '130',
```
Comment (lb-) [inline client/tailwind.config.js:87]: Ignore that. I see we are also updating the core tokens.

File: wagtail/admin/templates/wagtailadmin/shared/dialog/dialog-toggle.html [hunk 10]
```diff
@@ -0,0 +1,12 @@
+{% load wagtailadmin_tags %}
+{% comment %}
+    Variables this template accepts:
+
+    dialog_title - The title of the dialog you are toggling open (must match)
+    class_name - Css classes for styling the button
+    text - The text of the button
+
+{% endcomment %}
+<button type="button" class="{{ class_name }}" data-a11y-dialog-show="{{ dialog_title|cautious_slugify }}">
```
[the remark this answers is not in the document]
Comment (lb-) [reply wagtail/admin/templates/wagtailadmin/shared/dialog/dialog-toggle.html:10]: Should this maybe accept a dialog_id instead? And do the slugify when the template include is used instead.

File: wagtail/contrib/styleguide/templates/wagtailstyleguide/base.html [hunk 11]
```diff
@@ -652,6 +653,48 @@ <h2>Dropdown buttons</h2>
...
+                <div>
+                    {% dialog_toggle class_name='button button-primary' dialog_title='Dialog with critical error' text='Dialog with error' %}
+
+                    {% dialog icon_name="globe" title="Dialog with critical error" subtitle="This is a testing subtitle" message_status="critical" message_heading="There was an issue with the thing" message_description="This is a subtext for the message" %}
+                        <p class="w-base-text">This dialog message was generated by passing message_status=critical as well as message_heading and message_description to the dialog template tag</p>
+                    {% enddialog %}
+                </div>
+            </div>
+        </section>
+
```
Comment (lb-) [inline wagtail/contrib/styleguide/templates/wagtailstyleguide/base.html:697]: Are we adding stories as part of this PR also?

File: client/scss/components/_dialog.scss (hunk 4 above)
Comment (stevedya) [reply client/scss/components/_dialog.scss:137]: Oh actually Ben wanted this in grey, so that when we allow theming, the custom primary colour doesn't clash or become in-accessible with the bg.

Comment (stevedya) [reply client/scss/components/_dialog.scss:137]: Oh actually Ben wanted this in grey, so that when we allow theming, the custom primary colour doesn't clash or become in-accessible with the bg. I have updated the font size now however.

File: wagtail/contrib/styleguide/templates/wagtailstyleguide/base.html (hunk 11 above)
Comment (stevedya) [reply wagtail/contrib/styleguide/templates/wagtailstyleguide/base.html:697]: I think this will be part of a separate PR unfortunately :/

Comment (lb-) [reply wagtail/contrib/styleguide/templates/wagtailstyleguide/base.html:697]: Good call. Just wanted to check it was on the radar.

File: wagtail/admin/templates/wagtailadmin/shared/dialog/dialog.html [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+
+                {% if subtitle %}
+                    <p class="w-dialog__subtitle w-help-text">{{ subtitle }}</p>
+                {% endif %}
+
+
+                {% comment %}
+    This markup is intentionally left without closing div tags so that the contents can be populated with child elements between dialog and enddialog
+    For the end tags please see end-dialog.html
+                {% endcomment %}
```
Comment (lb-) [inline wagtail/admin/templates/wagtailadmin/shared/dialog/dialog.html:39]: Just wondering. If we put the closing divs all in one line within the comment - will curlylint be tricked into thinking the Dom structure is ok? Or maybe within a html comment. 

Or even within an if statement that will never be true?

Pretty hacky but just a thought.

Comment (lb-) [reply wagtail/admin/templates/wagtailadmin/shared/dialog/dialog.html:39]: Ok. Crazy idea. We use one template only and provide a context variable for is_start or similar and then put the whole html in one file?
